src/intents/TokenRegistry.ts

```diff
@@ -0,0 +1,70 @@
+type TokenInput = {
+  address: string;
+  decimals?: number;
+}
+
+type TokenMetadata = {
+  address: string;
+  symbol: string;
+  decimals: number;
+};
+
+type TokenReturn = {
+  address: string;
+  decimals: number;
+};
+
+export class TokenRegistry {
+  private addressMap: Map<string, TokenMetadata>;
```

Comment:
using maps is for faster lookups. idk how big these lists are.

maps is O1, arrays is O(n)